docs/examples/websocket.c

```diff
@@ -30,6 +30,39 @@
 #include <unistd.h>
 #include <curl/curl.h>
 
+/* Auxiliary function that waits on the socket. */
+static int wait_on_socket(curl_socket_t sockfd, long timeout_ms)
+{
+  struct timeval tv;
+  fd_set infd, outfd, errfd;
+  int res;
+
+  tv.tv_sec = timeout_ms / 1000;
+  tv.tv_usec = (int)(timeout_ms % 1000) * 1000;
+
+  FD_ZERO(&infd);
+  FD_ZERO(&outfd);
+  FD_ZERO(&errfd);
+
+/* Avoid this warning with pre-2020 Cygwin/MSYS releases:
+ * warning: conversion to 'long unsigned int' from 'curl_socket_t' {aka 'int'}
+ * may change the sign of the result [-Wsign-conversion]
+ */
+#if defined(__GNUC__) && defined(__CYGWIN__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#endif
+  FD_SET(sockfd, &errfd); /* always check for error */
+  FD_SET(sockfd, &infd);
+#if defined(__GNUC__) && defined(__CYGWIN__)
```

Comment:
```suggestion
#if defined(__GNUC__)
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wsign-conversion"
#endif
  FD_SET(sockfd, &errfd); /* always check for error */
  FD_SET(sockfd, &infd);
#if defined(__GNUC__)
```
This hack is also necessary on some non-Cygwin systems. (OmniOS for example)

Reply:
@vszakats I've forced pushed my changes.
I was just following the same pattern found in [docs/examples/sendrecv.c](https://github.com/curl/curl/blob/master/docs/examples/sendrecv.c), which now was changed too.

Reply:
Thank you @Theldus.